tailscale/resource_dns_split_nameservers.go

```diff
@@ -0,0 +1,112 @@
+package tailscale
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/tailscale/tailscale-client-go/tailscale"
+)
+
+func resourceDNSSplitNameservers() *schema.Resource {
+	return &schema.Resource{
```

Comment:
I'd suggest adding an importer as well to allow users to import existing Split DNS configuration to their Terraform state. I suspect the usual `&schema.ResourceImporter{StateContext: schema.ImportStatePassthroughContext}` would work here.